src/combat.mjs

```diff
@@ -1,23 +1,154 @@
 import {item_roll, Logger} from "./util.mjs";
 import {socket} from "./module.mjs";
-import {ID_MAP, MODULE_ID} from "./constants.mjs";
+import {COMBAT_HEALTH_ESTIMATE, ID_MAP, MODULE_ID} from "./constants.mjs";
+import { actor_to_discord_ids } from "./sync.mjs";
 
 
-//TODO: Include some logic to set/remove these hooks when combat_tracking_enabled is changed
 export function set_combat_hooks() {
     Logger.info("Setting Combat Hooks.")
 
     Hooks.on("combatStart", async (combat, updateData) => {
-        socket.emit('combat', combat.toJSON())
+        const roundRender = parseCombatRound({ ...combat, ...updateData })
+        const turnRender = parseTurn(combat, updateData) 
+        socket.emit('combat', roundRender+turnRender)
     })
-    Hooks.on("combatTurn", async (combat, updateData) => {
-        socket.emit('combat', combat.toJSON())
+    Hooks.on("combatTurn", async (combat, updateData, updateOptions) => {
+        if (updateOptions.direction < 1) return
+        const turnRender = parseTurn(combat, updateData) 
+        socket.emit('combat', turnRender)
     })
-    Hooks.on("combatRound", async (combat, updateData) => {
-        socket.emit('combat', combat.toJSON())
+    Hooks.on("combatRound", async (combat, updateData, updateOptions) => {
+        if (updateOptions.direction < 1) return
+        const roundRender = parseCombatRound({ ...combat, ...updateData }, updateOptions)
+        const turnRender = parseTurn(combat, updateData) 
+        socket.emit('combat', roundRender+turnRender)
     })
 }
 
+function getEffectsInMarkdown(actor, token) {
+    let markdown = ''
+    const effects = (token.actorlink) ? actor.effects : token.document.delta.effects
```

Comment:
Consider filtering out common effects. Some I noticed were:

- Jack of All Trades
- Prepared Spells
- Unarmored Defense
- Fast Movement
- Unarmored Movement
- Swim Speed
- Arcane Firearm

Reply:
I assumed this property was only used for combat effects.
Going to do a little testing on this to replicate and see if I can differentiate combat effects vs character traits.

Reply:
Tested and replaced the if statement with a merge since sometimes, for linked tokens, the actor and the token may be in sync.

I've tested a little and none of those conditions above should be displayed. Only conditions, temporary and active effects are shown in these place, not passive effects. Which I think is the right way, I don't think GMs care to communicate passive effects to the players.

![image](https://github.com/oronder/Oronder/assets/4229134/bef3641a-9848-43f9-9612-f5dec99e58ab)

Unarmored Defense is not shown in combat tracker:
![image](https://github.com/oronder/Oronder/assets/4229134/6dcb979a-6833-43c8-a6eb-6fde51296866)

There is still an issue here, I put rage on a barbarian and it's not showing up. This is a temporary effect but for some reason not showing up when other temporary effects are.

Reply:
Alright fixed.
It picks up active and temporary effects from the actor, items and features but ignores passives ones unless they have an attached status(i.e. flying, incapacitated etc).

![image](https://github.com/oronder/Oronder/assets/4229134/6ff0d71e-3267-482c-a452-3fab614d7ba4)
![image](https://github.com/oronder/Oronder/assets/4229134/d1fef7b8-dd78-4eea-8ff4-a6c0fce61c16)

Reply:
I think this issue is good to be resolved if you are @ChunkLightTuna